core/src/smartcontracts/isi/query.rs

```diff
@@ -33,33 +33,12 @@ impl ValidQueryRequest {
     }
 }
 
-/// Unsupported version error
-#[derive(Clone, Copy, Eq, PartialEq, Debug, Decode, Encode, IntoSchema, Error)]
-#[error(
-    "Unsupported version. Expected: {}, got: {version}",
-    Self::expected_version()
-)]
-pub struct UnsupportedVersionError {
-    /// Version that we got
-    pub version: u8,
-}
-
-impl UnsupportedVersionError {
-    /// Expected version
-    pub const fn expected_version() -> u8 {
-        1
-    }
-}
-
 /// Query errors.
-#[derive(Error, Debug, Clone, Decode, Encode, IntoSchema)]
```

Comment:
Not sure if it's a good idea: see this [guideline](https://rust-lang.github.io/api-guidelines/interoperability.html#c-common-traits)
From the other side we can always bring it back if we need it.

Me personally always implement `Debug` and `Clone` for all types

Reply:
Yes, we can impl `Clone` when needed